File: 프로그래머스/lv1/131128. 숫자 짝꿍/숫자 짝꿍.java
class Solution {
    public static String solution(String X, String Y) {
		int[] x = new int[10];
		int[] y = new int[10];
		StringBuilder answer = new StringBuilder();

		for (int i = 0; i < X.length(); i++) {
			x[Character.getNumericValue(X.charAt(i))]++;
		}

		for (int i = 0; i < Y.length(); i++) {
			y[Character.getNumericValue(Y.charAt(i))]++;
		}

		for(int i = 9; i >= 0; i--){
			answer.append(String.valueOf(i).repeat(Math.max(0, Math.min(x[i], y[i]))));
		}

		if (answer.toString().length() == 0) return "-1";
		if (answer.toString().charAt(0) == '0') return "0";

		return answer.toString();
	}
}

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.kt
import kotlin.math.max
import kotlin.math.min

class Solution {
    fun solution(X: String, Y: String): String {
       val x = IntArray (10)
        val y = IntArray (10)

        X.forEach { x[it.digitToInt()]++ }
        Y.forEach { y[it.digitToInt()]++ }

        val ans = StringBuilder()
        for (i in 9 downTo 0)
            ans.append(i.toString().repeat(max(0, min(x[i], y[i]))))

        if (ans.isEmpty()) return "-1"
        if (ans.startsWith("0")) return "0"

        return ans.toString()
    }
}